src/PrestaShopBundle/Resources/config/services/bundle/form/form_type.yml

```diff
@@ -626,6 +626,8 @@ services:
         class: 'PrestaShopBundle\Form\Admin\Catalog\Category\CategoryType'
         parent: 'form.type.catalog.abstract_category'
         public: true
+        arguments:
+          - '@=service("prestashop.adapter.legacy.configuration").getInt("PS_ROOT_CATEGORY")'
```

Comment:
I think you don't need to inject this, all `CommonAbstractType` children have the protected method `getConfiguration`

Reply:
but is it good to use configuration from `CommonAbstractType`, which is instantiated there and not loaded from container? I think it will be refactored in the future as it's not good design https://github.com/PrestaShop/PrestaShop/blob/develop/src/PrestaShopBundle/Form/Admin/Type/CommonAbstractType.php#L44

Reply:
It's bad design I agree with you, but since the service has no parameters it works ^^
We will improve this later via a setter injection probably, but for now you can use `getConfiguration`

Reply:
fixed it :)